db/src/urls.rs

```diff
@@ -136,12 +145,31 @@ async fn store_markdown(
     let category_result = store_category(pool, markdown_result.markdown_id, &category).await?;
     println!("Category result: {category_result:?}");
 
-    Ok(markdown_result)
+    Ok((markdown_result, markdown_content))
 }
 
-async fn store_raw_html_in_page_snapshot(
+pub async fn generate_and_store_summary(
+    pool: &PgPool,
+    markdown_id: Uuid,
+    cleaned_html: &str,
+) -> color_eyre::Result<()> {
+    let summary = generate_summary(cleaned_html).await?;
+
+    sqlx::query!(
+        "UPDATE Markdown SET summary = $1 WHERE markdown_id = $2",
+        summary,
+        markdown_id
+    )
+    .execute(pool)
+    .await?;
+
+    Ok(())
+}
+
+pub async fn persist_article(
     pool: &PgPool,
     page: Page,
+    _markdown: Markdown,
```

Comment:
Do we need to be passing down this `Markdown`? Aren't we making the markdown inside this method?

Reply:
I thought I would have to pass down to `store_markdown_article`. I think all these silly mistakes are caused because the `db/url.rs` code is kind of all over the place. I will make sure to clean it up